experiments/experiments_test.go

```diff
@@ -0,0 +1,151 @@
+package experiments
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	parser "github.com/Netflix/p2plab/cue/parser"
+	"github.com/Netflix/p2plab/metadata"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExperimentDefinition(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	db, cleanup := newTestDB(t, "exptestdir")
+	defer func() {
+		if err := cleanup(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	var ids []string
+	t.Run("Experiment Creation And Retrieval", func(t *testing.T) {
+		sourceFiles := []string{
+			"../cue/cue.mod/p2plab_example1.cue",
+			"../cue/cue.mod/p2plab_example2.cue",
+		}
+		for _, sourceFile := range sourceFiles {
+			name := strings.Split(sourceFile, "/")
+			exp1 := newTestExperiment(t, sourceFile, name[len(name)-1])
+			ids = append(ids, exp1.ID)
+			exp2, err := db.CreateExperiment(ctx, exp1)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
Still needs `require.NoError(t, err)`.

Reply:
ah shoot my bad, i will apply the requires to everything else